cmake/usMacroCreateBundle.cmake

```diff
@@ -5,7 +5,7 @@ macro(usMacroCreateBundle _project_name)
 cmake_parse_arguments(${_project_name}
   "SKIP_EXAMPLES;SKIP_INIT"
   "VERSION;TARGET;SYMBOLIC_NAME;EMBED_RESOURCE_METHOD"
-  "DEPENDS;PRIVATE_INCLUDE_DIRS;LINK_LIBRARIES;SOURCES;PRIVATE_HEADERS;PUBLIC_HEADERS;RESOURCES;BINARY_RESOURCES"
+  "DEPENDS;PRIVATE_INCLUDE_DIRS;LINK_LIBRARIES;SOURCES;PRIVATE_HEADERS;PUBLIC_HEADERS;RESOURCES;BINARY_RESOURCES;BUILD_OBJ"
```

Comment:
Making symbols visible to the test code this way is not what we want to do. I believe this will expose any class/function/symbol tagged with `US_ABI_TEST` macro as a public symbol when we ship this library and that is not what we want. I think the best way is to split the cppms core library CMake target into two targets; one that builds all the object files and one that links all the object files into the final shipping shared library. The test bed will link against the object files and not the final shipping shared library.

Reply:
You're right sorry, I think a commit got lost on my way out since I got lazy with naming my stashes...  Just recreated it, let me know if it's satisfactory.

The way I ended up approaching it was to add a second framework target that defines `US_ABI_TESTING`, then linking only the tests against it.  That way the final `libCppMicroServicesd.so` isn't affected at all, but none of the linking configuration has to change.  I tried the object file approach for a long time, but ultimately it broke too many things in weird ways due to the global bundle context for `system_bundle` (especially on Windows, which allows symbols with duplicate names in different places to refer to different things).  It was much simpler to just redefine the macros responsible for exporting symbols and produce two separate targets.